apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx

```diff
@@ -39,21 +39,27 @@ import {
 } from './utils'
 import { useParams } from 'react-router-dom'
 import { PathParams } from '../../RouteDefinitions'
-import { usePullRequestData } from './context/pull-request-data-provider'
 import { isEmpty } from 'lodash-es'
 import { CodeOwnerReqDecision } from '../../types'
 import { useAppContext } from '../../framework/context/AppContext'
+import { usePullRequestDataStore } from './stores/pull-request-store'
 
 export default function PullRequestConversationPage() {
   const {
-    refetchActivities,
-    refetchPullReq,
     pullReqMetadata,
-    pullReqChecksDecision,
-    prPanelData,
+    refetchPullReq,
+    refetchActivities,
     setRuleViolationArr,
-    loading: prLoading
-  } = usePullRequestData()
+    prPanelData,
+    pullReqChecksDecision
+  } = usePullRequestDataStore(state => ({
+    pullReqMetadata: state.pullReqMetadata,
+    refetchPullReq: state.refetchPullReq,
+    refetchActivities: state.refetchActivities,
+    setRuleViolationArr: state.setRuleViolationArr,
+    prPanelData: state.prPanelData,
+    pullReqChecksDecision: state.pullReqChecksDecision
+  }))
   const { currentUser: currentUserData } = useAppContext()
   const [checkboxBypass, setCheckboxBypass] = useState(false)
```

Comment:
are we planning to move these `useStates` to zustand as well separately?

Reply:
I was planning to move the states to the zustand store. I didn't want to do it all in one pr tho ahah